openwisp_monitoring/monitoring/base/models.py

```diff
@@ -205,6 +221,14 @@ def content_type_key(self):
         except AttributeError:
             return None
 
+    @property
+    def alert_field(self):
+        return self.config_dict.get('alert_field', self.field_name)
```

Comment:
If the `alert_field` is configured for the metric in the project's setting as shown below: 

```python
OPENWISP_MONITORING_METRICS = {
    'iperf': {
        'alert_field': 'iperf_result',
    }
}
```

Then, changing the alert field from the AlertInline on the device change admin will have no effect. The "Field name" in the AlertInline represents `metric.field_name`. 

I think we can solve this problem by modifying this code as follows

```python
    @property
    def alert_field(self):
        if self.field_name != self.config_dict['field_name']:
            return self.field_name
        return self.config_dict.get('alert_field', self.field_name)
```

Reply:
Will update it, thanks!